python/pyspark/worker.py

```diff
@@ -582,6 +583,56 @@ def read_udtf(pickleSer, infile, eval_type):
             message_parameters={"method_name": "__init__", "error": str(e)},
         )
 
+    """
+    This implements the logic of a UDTF that accepts an input TABLE argument with one or more
+    PARTITION BY expressions.
+
+    Parameters
+    ----------
+    create_udtf: function
+        Function to create a new instance of the UDTF to be invoked.
+    partition_child_indexes: list
+        List of integers identifying zero-based indexes of the columns of the input table that
+        contain projected partitioning expressions. This class will inspect these values for each
+        pair of consecutive input rows. When they change, this indicates the boundary between two
+        partitions, and we will invoke the 'terminate' method on the UDTF class instance and then
+        destroy it and create a new one to implement the desired partitioning semantics.
+    """
+    class UDTFWithPartitions:
```

Comment:
nit: In Python, a class comment is placed after `class Xxxx:`:

```py
class UDTFWithPartitions:
    """
    This implements ...
    """

    def __init__( ... ):
...
```

Reply:
Sounds good, done.